src/apiutils.jl

```diff
@@ -55,36 +55,31 @@ end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x,
                seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
-    for i in eachindex(duals)
-        duals[i] = Dual{T,V,N}(x[i], seed)
-    end
+    duals .= Dual{T,V,N}.(x, (seed,))
     return duals
 end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x,
                seeds::NTuple{N,Partials{N,V}}) where {T,V,N}
-    for i in 1:N
-        duals[i] = Dual{T,V,N}(x[i], seeds[i])
-    end
+    i = 1:N
+    @views duals[i] .= Dual{T,V,N}.(x[i], seeds)
```

Comment:
Perhaps these `@views` could be reduced in "strength" by putting them where they are needed, e.g. only on the `x[i]`? And could the usage of `@.` be consistent among the changes (it's not used here, but used in other places).